src/daemon/dlt_daemon_common.c

```diff
@@ -141,6 +141,92 @@ DltDaemonRegisteredUsers *dlt_daemon_find_users_list(DltDaemon *daemon,
     dlt_vlog(LOG_ERR, "Cannot find user list for ECU: %4s\n", ecu);
     return (DltDaemonRegisteredUsers *)NULL;
 }
+
+#ifdef DLT_LOG_LEVEL_APP_CONFIG
+
+static int dlt_daemon_cmp_log_settings(const void *lhs, const void *rhs) {
+    if ((lhs == NULL) || (rhs == NULL))
+        return -1;
+
+    DltDaemonContextLogSettings *settings1 = (DltDaemonContextLogSettings *)lhs;
+    DltDaemonContextLogSettings *settings2 = (DltDaemonContextLogSettings *)rhs;
+
+    int cmp = memcmp(settings1->apid, settings2->apid, DLT_ID_SIZE);
+
+    if (cmp < 0)
+        return -1;
+    else if (cmp == 0)
+        return memcmp(settings1->ctid, settings2->ctid, DLT_ID_SIZE);
+    else
+        return 1;
+}
+
+/**
+ * Find configuration for app/ctx id specific log settings configuration
+ * @param daemon pointer to dlt daemon struct
+ * @param apid application id to use
+ * @param ctid context id to use, can be NULL
+ * @return pointer to log settings if found, otherwise NULL
+ */
+DltDaemonContextLogSettings *dlt_daemon_find_configured_app_id_ctx_id_settings(
+    const DltDaemon *daemon, const char *apid, const char *ctid) {
+  DltDaemonContextLogSettings *app_id_settings = NULL;
+    for (int i = 0; i < daemon->num_app_id_log_level_settings; ++i) {
+      DltDaemonContextLogSettings *settings = &daemon->app_id_log_level_settings[i];
+
+        if (strncmp(apid, settings->apid, DLT_ID_SIZE) != 0) {
+            if (app_id_settings != NULL)
+                return app_id_settings;
+            continue;
+        }
+
+        if (strlen(settings->ctid) == 0) {
+            app_id_settings = settings;
+        }
+
+        if (ctid == NULL || strlen(ctid) == 0) {
+            if (app_id_settings != NULL) {
+                return app_id_settings;
+            }
+        } else {
+            if (strncmp(ctid, settings->ctid, DLT_ID_SIZE) == 0) {
+                return settings;
+            }
+        }
+    }
+
+    return app_id_settings;
+}
+
+/**
+ * Find configured log levels in a given DltDaemonApplication for the passed context id.
+ * @param app The application settings which contain the previously loaded ap id settings
+ * @param ctid The context id to find.
+ * @return Pointer to DltDaemonApplicationLogSettings containing the log level
+ *         for the requested application or NULL if none found.
+ */
+DltDaemonContextLogSettings *dlt_daemon_find_app_log_level_config(
+    const DltDaemonApplication *const app, const char *const ctid) {
+
+  if (NULL == ctid)
+    return NULL;
+
+  DltDaemonContextLogSettings settings;
+  memcpy(settings.apid, app->apid, DLT_ID_SIZE);
+  memcpy(settings.ctid, ctid, DLT_ID_SIZE);
+
+  DltDaemonContextLogSettings* log_settings = NULL;
+  log_settings =
+      (DltDaemonContextLogSettings *)bsearch(
+          &settings, app->context_log_level_settings,
+          (size_t)app->num_context_log_level_settings,
+          sizeof(DltDaemonContextLogSettings),
+          dlt_daemon_cmp_log_settings);
+  return log_settings;
+}
```

Comment:
Please check alignment for whole method dlt_daemon_find_app_log_level_config.

Reply:
done